### PR TITLE
dbutil add snapshot logic

### PR DESCRIPTION
**Pull request by egonelbre**

### CODE REVIEW

File: internal/dbutil/pgutil/data.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	"strconv"
+	"strings"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/internal/dbutil/dbschema"
+)
+
+// QueryData loads all data from tables
+func QueryData(db Queryer, schema *dbschema.Schema) (*dbschema.Data, error) {
```
Comment (mniewrzal) [inline internal/dbutil/pgutil/data.go]: hmm everything except `quote_nullable` look to be independent from `postgres`. Maybe it can moved to `dbutil`?

Comment (egonelbre) [reply internal/dbutil/pgutil/data.go]: Hmm, I could move it into `dbschema with an optional `quoteColumn func(string)` ?

Comment (mniewrzal) [reply internal/dbutil/pgutil/data.go]: yes, I think thats easiest solution.

Comment (egonelbre) [reply internal/dbutil/pgutil/data.go]: done

File: internal/dbutil/pgutil/data.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	"strconv"
+
+	"storj.io/storj/internal/dbutil/dbschema"
+)
+
+// QueryData loads all data from tables
+func QueryData(db dbschema.Queryer, schema *dbschema.Schema) (*dbschema.Data, error) {
+	return dbschema.QueryData(db, schema, func(columnName string) string {
+		quoted := strconv.Quote(columnName)
+		return `quote_nullable(` + quoted + `) as ` + quoted
```
Comment (wthorp) [inline internal/dbutil/pgutil/data.go]: I think strconv.Quote + quote_nullable will double-escape things

File: internal/dbutil/pgutil/db.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+func (db *DB) Close() error {
+	return errs.Combine(
+		DropSchema(db.DB, db.Schema),
+		db.DB.Close(),
+	)
+}
+
+// LoadSchemaFromSQL inserts script into connstr and loads schema.
+func LoadSchemaFromSQL(connstr, script string) (_ *dbschema.Schema, err error) {
+	db, err := Open(connstr, "load-schema")
```
Comment (wthorp) [inline internal/dbutil/pgutil/db.go]: does using the fixed 'load-schema' schema name here have data race implications?

File: internal/dbutil/dbschema/data.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+			defer func() { err = errs.Combine(err, rows.Close()) }()
+
+			row := make(RowData, len(columnNames))
+			rowargs := make([]interface{}, len(columnNames))
+			for i := range row {
+				rowargs[i] = &row[i]
+			}
+
+			for rows.Next() {
+				err := rows.Scan(rowargs...)
```
Comment (wthorp) [inline internal/dbutil/dbschema/data.go]: I need to brush up on my variadics.  Does this work, or should rowargs be []*interface{} instead?
"If an argument has type *interface{}, Scan copies the value provided by the underlying driver without conversion"

File: internal/dbutil/pgutil/db.go (hunk 3 above)
Comment (egonelbre) [reply internal/dbutil/pgutil/db.go]: no `Open` always randomizes the full schema name.

Comment (wthorp) [reply internal/dbutil/pgutil/db.go]: ah yes

File: internal/dbutil/pgutil/data.go (hunk 2 above)
Comment (egonelbre) [reply internal/dbutil/pgutil/data.go]: `strconv.Quote` escapes the column name not the value. `quote_nullable` escapes the value retrieved from the database.

File: internal/dbutil/dbschema/data.go (hunk 4 above)
Comment (egonelbre) [reply internal/dbutil/dbschema/data.go]: I do know that the resulting value is a string, so I provide an string pointer as a place to store the data. If I didn't know the actual type, then yes, I would need to use `*interface{}` instead.

**Pull request merged**